analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+---
+title: "Clustering SCPCL000822 and SCPCL000824"
+author: Ally Hawkins
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: true
+    toc_depth: 3
+    code_folding: "hide"
+---
+
+This notebook does some initial exploration of clustering in two libraries from `SCPCP000015`: `SCPCL000822` and `SCPCL000824`. 
+
+- Louvain clusters are calculated using a range of parameters specifying nearest neighbors (5, 10, 15, 20, 25, 30, 35, and 40). 
+- Metrics are then calculated to evaluate clustering results across all parameters: 
+  - Silhouette width 
+  - Cluster purity 
+  - Cluster stability 
+  
+  
+The clusters are then compared to the results from running `SingleR` in the `aucell-singler-annotation.sh` workflow.
+We then look at marker gene expression across the clusters and assign each cluster to a cell type. 
+
+## Setup
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+
+# Set default ggplot theme
+theme_set(
+  theme_bw()
+)
+
+# set seed
+set.seed(2024)
+```
+
+
+```{r base paths}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current data directory, found within the repository base directory
+data_dir <- file.path(repository_base, "data", "current", "SCPCP000015")
+
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-ewings")
+
+# path to marker genes file
+marker_genes_file <- file.path(module_base, "references", "visser-all-marker-genes.tsv")
+```
+
+```{r}
+# source in helper functions for make_jaccard_matrix() and jaccard()
+jaccard_functions <- file.path(module_base, "scripts", "utils", "jaccard-functions.R")
+source(jaccard_functions)
+
+# source in helper functions: plot_gene_heatmap() and plot_cnv_heatmap()
+# create_classification_df() and create_marker_gene_df()
+validation_functions <- file.path(module_base, "scripts", "utils", "tumor-validation-helpers.R")
+source(validation_functions)
+```
+
+## Define functions 
+
+```{r, assign_clusters_functions}
+# get clusters for a specified value of k
```

Comment:
Expand this "docs" a bit here that it's louvain w/ jaccard since `k` means many different things in clustering land!